docs-en/chapter_data_structure/classification_of_data_structure.md

```diff
@@ -20,23 +21,28 @@ Non-linear data structures can be further divided into tree structures and netwo
+**During the execution of an algorithm, the data being processed is stored in memory**. The figure below shows a computer memory stick where each black square represents a memory space. We can imagine memory as a vast Excel spreadsheet in which each cell can store data of a specific size.
+
+**The system accesses the data at the target location by means of a memory address**. As shown in the figure below, the computer assigns a unique identifier to each cell in the table according to specific rules, ensuring that each memory space has a unique memory address. With these addresses, the program can access the data stored in memory.
+
 
 ![Memory Stick, Memory Spaces, Memory Addresses](classification_of_data_structure.assets/computer_memory_location.png)
 
 !!! tip
 
-    It's worth noting that comparing memory to an Excel spreadsheet is a simplified analogy. The actual working mechanism of memory is more complex, involving concepts like address space, memory management, cache mechanisms, virtual memory, and physical memory.
 
-Memory is a shared resource for all programs. When a block of memory is occupied by one program, it cannot be used by others simultaneously. **Therefore, memory resources are an important consideration in the design of data structures and algorithms**. For example, the peak memory usage of an algorithm should not exceed the system's remaining free memory. If there is a lack of contiguous large memory spaces, the chosen data structure must be able to store data in dispersed memory spaces.
+    It is worth noting that comparing memory to an Excel spreadsheet is a simplified analogy. The actual memory working mechanism is more complex, involving concepts such as address, space, memory management, cache mechanism, virtual and physical memory.
+
+Memory is a shared resource for all programs. When a block of memory is occupied by one program, it cannot be simultaneously used by other programs. **Therefore, considering memory resources is crucial in designing data structures and algorithms**. For instance, the algorithm's peak memory usage should not exceed the remaining free memory of the system; if there is a lack of contiguous memory blocks,  then the data structure chosen must be able to be stored in non-contiguous memory blocks.
+
+As illustrated in the figure below, **the physical structure reflects the way data is stored in computer memory and it can be divided into contiguous space storage (arrays) and non-contiguous space storage (linked lists)**. This physical structure determines how data is accessed, updated, added, deleted, etc. Logical and physical structure complement each other in terms of time efficiency and space efficiency.
```

Comment:
Logical and physical structure complement each other in terms of time efficiency and space efficiency -> Contiguous and non-contiguous structures complement each other in terms of time efficiency and space efficiency

I think the original text meant differently, please check the Chinese version again to make sure this is correct.

Reply:
Also, the bolded text is not consistent with the original text, please change it back if possible. It should end before "and it can be divided ..."

Reply:
Hi Yuelinxin
thank you so much for your review and tips, I have tried playing around with the editor and fixed the issue of trouble seeing extra spaces. Sorry for my inadvertence. I will fix them up now!